SpatialGDK/Source/SpatialGDK/Public/Tests/SpatialView/ExpectedMessagesToSend.h

```diff
@@ -11,16 +11,17 @@ namespace SpatialGDK
 class ExpectedMessagesToSend
 {
 public:
-	ExpectedMessagesToSend& AddCreateEntityRequest(Worker_RequestId RequestId, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
-												   double ComponentValue);
-	ExpectedMessagesToSend& AddEntityCommandRequest(Worker_RequestId RequestId, Worker_EntityId EntityId);
+	ExpectedMessagesToSend& AddCreateEntityRequest(Worker_RequestId RequestId, Worker_EntityId EntityId,
+												   TArray<ComponentData> ComponentData);
+	ExpectedMessagesToSend& AddEntityCommandRequest(Worker_RequestId RequestId, Worker_EntityId EntityId, Worker_ComponentId ComponentId,
+													Worker_CommandIndex CommandIndex);
 	ExpectedMessagesToSend& AddDeleteEntityCommandRequest(Worker_RequestId RequestId, Worker_EntityId EntityId);
 	ExpectedMessagesToSend& AddReserveEntityIdsRequest(Worker_RequestId RequestId, uint32 NumOfEntities);
-	ExpectedMessagesToSend& AddEntityQueryRequest(Worker_RequestId RequestId);
+	ExpectedMessagesToSend& AddEntityQueryRequest(Worker_RequestId RequestId, EntityQuery Query);
 	ExpectedMessagesToSend& AddEntityCommandResponse(Worker_RequestId RequestId, Worker_ComponentId ComponentId,
 													 Worker_CommandIndex CommandIndex);
 	ExpectedMessagesToSend& AddEntityCommandFailure(Worker_RequestId RequestId, FString Message);
-	bool Compare(const TUniquePtr<MessagesToSend> MessagesToSend) const;
+	bool Compare(TUniquePtr<MessagesToSend> MessagesToSend) const;
```

Comment:
I'd just compare to const MessagesToSend& rather than to a unique pointer. There is no need to take ownership here.

Reply:
This might be a good reason to revert the union type in messages to send to be more like the original one in the worker wrapper repo though.